### PR TITLE
fix(suite-native): fix date of undefined error

### LINKED ISSUE #11917: TypeError: Cannot read property 'date' of undefined
Sentry Issue: [SUITE-NATIVE-MZ](https://satoshilabs.sentry.io/issues/5137682656/?referrer=github_integration)

```
TypeError: Cannot read property 'date' of undefined
  at setFingerPoint (/home/expo/workingdir/build/suite-native/react-native-graph/src/AnimatedLineGraph.tsx:358:69)
```

### PR DESCRIPTION
**Pull request by juriczech**
## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/11917

## Screenshots:
image

### CODE REVIEW

File: suite-native/react-native-graph/src/AnimatedLineGraph.tsx [hunk 1]
```diff
@@ -355,12 +355,10 @@ export function AnimatedLineGraph<TEventPayload extends object>({
         (fingerX: number) => {
             const fingerXInRange = Math.max(fingerX - horizontalPadding, 0);
 
-            const lastDate = pointsInRange[pointsInRange.length - 1].date;
-
             // If there was graph error, points in range can be empty
-            if (!lastDate) {
-                return;
-            }
+            if (pointsInRange.length === 0) return;
```
Comment (Nodonisko) [inline suite-native/react-native-graph/src/AnimatedLineGraph.tsx]: If you suggest this:

```tsx
const lastDate = pointsInRange[pointsInRange.length - 1]?.date;

if(!lastDate) return;
```
 
This version is little bit safer and future proof and will also pass https://www.totaltypescript.com/tips/make-accessing-objects-safer-by-enabling-nouncheckedindexedaccess-in-tsconfig (we sadly don't have this enabled)

Comment (Nodonisko) [reply suite-native/react-native-graph/src/AnimatedLineGraph.tsx]: maybe you can try to turn it on for this package if it won't produce too many errors

Comment (juriczech) [reply suite-native/react-native-graph/src/AnimatedLineGraph.tsx]: [6635ff5](https://github.com/trezor/trezor-suite/pull/11924/commits/6635ff5558831c8b861b8e5972d70e63efe107a9)

**Review by Nodonisko — APPROVED** [review]

**Pull request merged**